control/veda_deliver_cielo.py

```diff
@@ -1,186 +1,228 @@
+    """
+    pass
 
-requests.packages.urllib3.disable_warnings()
 
+class Cielo24CreateJobError(Cielo24Error):
+    """
+    An error occurred during new job creation.
+    """
+    pass
 
-class Cielo24Transcript():
-
-    def __init__(self, veda_id):
-        self.veda_id = veda_id
-        '''Defaults'''
-        self.c24_site = 'https://api.cielo24.com/api'
-        self.c24_login = '/account/login'
-        self.c24_joblist = '/job/list'
-        self.c24_newjob = '/job/new'
-        self.add_media = '/job/add_media'
-        self.transcribe = '/job/perform_transcription'
-
-        '''Retreive C24 Course-based defaults'''
-        self.c24_defaults = self.retrieve_defaults()
-
-    def perform_transcription(self):
-        if self.c24_defaults['c24_user'] is None:
-            return None
-        '''
-        GET /api/job/perform_transcription?v=1 HTTP/1.1
-        &api_token=xxxx
-        &job_id=xxxx
-        &transcription_fidelity=PREMIUM&priority=STANDARD
-        Host: api.cielo24.com
-        '''
-        api_token = self.tokengenerator()
-        if api_token is None:
-            return None
-
-        job_id = self.generate_jobs(api_token)
-        task_id = self.embed_url(api_token, job_id)
-
-        r5 = requests.get(
-            ''.join((
-                self.c24_site,
-                self.transcribe,
-                '?v=1&api_token=',
-                api_token,
-                '&job_id=',
-                job_id,
-                '&transcription_fidelity=',
-                self.c24_defaults['c24_fidelity'],
-                '&priority=',
-                self.c24_defaults['c24_speed']
-            ))
-        )
-        return ast.literal_eval(r5.text)['TaskId']
 
-    def retrieve_defaults(self):
-        video_query = Video.objects.filter(
-            edx_id=self.veda_id
-        ).latest()
+class Cielo24AddMediaError(Cielo24Error):
+    """
+    An error occurred during add media.
+    """
+    pass
+
 
-        if video_query.inst_class.mobile_override is True:
+class Cielo24PerformTranscriptError(Cielo24Error):
+    """
+    An error occurred during perform transcript.
+    """
+    pass
+
+
+class Cielo24Transcript(object):
+    """
+    Cielo24 Integration
+    """
+    def __init__(self, video, video_transcript_preferences):
+        self.video = video
+        self.video_transcript_preferences = video_transcript_preferences
+        # Defaults
+        self.cielo24_site = 'https://api.cielo24.com/api'
+        self.cielo24_new_job = '/job/new'
+        self.cielo24_add_media = '/job/add_media'
+        self.cielo24_perform_transcription = '/job/perform_transcription'
+        self.callback_base_url = reverse('cielo24_transcript_completed')
+
+        # Set Cielo24 defaults
+        self.cielo24_defaults = self.set_defaults()
+
+    def set_defaults(self):
+        """
+        Set default values for cielo24.
+        """
+        if self.video.inst_class.mobile_override is True:
```

Comment:
Last updated URL record with `_LBO` suffix was on Jan. 26, 2017, 6:33 a.m. So, it sounds like we don't use this profile anymore and since, 3rd party feature is for upcoming video uploads, we can always go for `_DTH.mp4`.

@yro may you please correct me on this one? thanks!